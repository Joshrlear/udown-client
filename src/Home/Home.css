#map {
    height: 100%;
    position: relative;
}

.map_container {
    height: 93vh;
    width: 100vw;
    margin: 0;
    padding: 0;
}
