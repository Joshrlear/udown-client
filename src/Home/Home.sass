.map_container 
    height: 100vh - 7vh
    width: 100vw
    margin: 0
    padding: 0


#map 
    height: 100%
    width: 100%
