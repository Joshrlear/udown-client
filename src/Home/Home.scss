@mixin desktop() {
  @media screen and ( min-width: 1100px ) {
    @content;
  }
}

@mixin tablet() {
  @media screen and ( min-width: 760px ) {
    @content;
  }
}

@mixin l-tablet() {
  @media screen and ( min-width: 900px ) {
    @content;
  }
}

#map {
  height: 100%;
  width: 100%;
}

.mapElement {
  position: relative;
  @include desktop() {
    position: absolute;
  }
}