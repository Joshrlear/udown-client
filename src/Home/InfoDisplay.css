body {
  margin: 0px;
}

img {
  -o-object-fit: cover;
     object-fit: cover;
  width: 100%;
  margin: 0px;
}

.info_display {
  position: absolute;
  overflow: auto;
  width: 100vw;
  height: 0vh;
  background-color: #F5F5F5;
  border-left: solid 1px lightgray;
}

.info_display.active {
  height: 65vh;
}

@media screen and (min-width: 1100px) {
  .info_display.active {
    height: 92vh;
    width: 45%;
    bottom: 0px;
    right: 0px;
  }
}

@media screen and (min-width: 1100px) {
  .info_container {
    display: -ms-grid;
    display: grid;
    position: relative;
    top: 5vh;
    grid: "info" "buttons";
  }
}

.info_container .info {
  width: 100%;
  padding: 0% 10% 10% 10%;
}

@media screen and (min-width: 1100px) {
  .info_container .info {
    display: -ms-grid;
    display: grid;
    grid-area: info;
  }
}

.info_container .info .review_header {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.info_container .info header > * {
  width: 100%;
  margin-top: 20px;
  text-align: center;
}

.info_container .button_container {
  margin-top: 20px;
}

@media screen and (min-width: 1100px) {
  .info_container .button_container {
    width: 40%;
    -webkit-box-pack: space-evenly;
        -ms-flex-pack: space-evenly;
            justify-content: space-evenly;
    grid-area: buttons;
  }
}

.info_container .button_container > button {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  width: 130px;
  height: 40px;
  margin: 0px;
}

.info_container .main_info {
  display: block;
  text-align: center;
  margin: 10px;
}

.info_container .review_list {
  width: 300px;
  margin: 0px;
  padding: 0px;
}

.info_container .address {
  position: relative;
  width: 200px;
  margin-top: 20px;
  right: 0px;
  width: 100%;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: (1fr)[7];
      grid-template-columns: repeat(7, 1fr);
}

.info_container .address h3, .info_container .address p {
  -ms-grid-column: 1;
  -ms-grid-column-span: 7;
  grid-column: 1/8;
}

@media screen and (min-width: 1100px) {
  .info_container .address h3, .info_container .address p {
    -ms-grid-column: 3;
    -ms-grid-column-span: 3;
    grid-column: 3/6;
  }
}
/*# sourceMappingURL=InfoDisplay.css.map */