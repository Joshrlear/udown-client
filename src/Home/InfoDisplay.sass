body 
    margin: 0px

  
img 
  object-fit: cover
  width: 100%
  margin: 0px


.info_display
  position: absolute
  overflow: auto
  width: 100vw
  height: 0vh
  background-color: #F5F5F5
  border-left: solid 1px lightgray
  transition: all .25s ease-out

  &.active
    height: 93vh - 35vh

.info_container
  

  .info 
    width: 100%
    padding: 0% 10% 10% 10%

    .review_header 
      display: flex


    header > * 
      width: 100%
      margin-top: 20px
      text-align: center

  .button_container
    margin-top: 20px

  .button_container > button 
      display: flex
      justify-content: space-around
      width: 130px
      height: 40px
      margin: 0px

  .main_info 
    display: block
    text-align: center
    margin: 10px


  .review_list
    width: 300px
    margin: 0px
    padding: 0px


  .address 
    position: relative
    width: 200px
    margin-top: 20px
    right: 0px
