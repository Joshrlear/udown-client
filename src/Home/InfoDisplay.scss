@mixin desktop() {
  @media screen and ( min-width: 1024px ) {
    @content;
  }
}

@mixin tablet() {
  @media screen and ( min-width: 760px ) {
    @content;
  }
}

@mixin l-tablet() {
  @media screen and ( min-width: 900px ) {
    @content;
  }
}
  
body {
    margin: 0px;
}
  
img {
  object-fit: cover;
  width: 100%;
  margin: 0px;
}

.info_display {
  position: absolute;
  overflow: auto;
  width: 100vw;
  height: 0vh;
  background-color: #F5F5F5;
  border-left: solid 1px lightgray;
  

  &.active {
    height: 100vh - 35vh;
    @include desktop() {
      height: 100vh - 8vh;
      width: 45%;
      bottom: 0px;
      right: 0px;
    }
  }
}
.info_container {
  @include desktop() {
    display: grid;
    position: relative;
    top: 5vh;
    grid: "info" "buttons";
  }

  .info {
    width: 100%;
    padding: 0% 10% 10% 10%;
    @include desktop() {
      display: grid;
      grid-area: info;
    }

    .review_header {
      display: flex;
    }

    header > * {
      width: 100%;
      margin-top: 20px;
      text-align: center;
    }
  }

  .button_container{
    margin-top: 20px;
    @include desktop() {
      width: 40%;
      grid-area: buttons;
    }
  }

  .button_container > button {
    width: 130px;
    height: 40px;
    margin-left: auto;
    margin-right: auto;
  }
  .main_info { 
    display: block;
    text-align: center;
    margin: 10px;
  }

  .review_list {
    width: 300px;
    margin: 0px;
    padding: 0px;
  }

  .address {
    position: relative;
    width: 200px;
    margin-top: 20px;
    right: 0px;
    width: 100%;
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    
    h3, p {
      grid-column: 1/8;
      @include desktop() {
        grid-column: 3/6;
      }
    }
  }
}