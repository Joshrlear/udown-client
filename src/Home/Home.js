import React, { Component, useState, useRef } from 'react';
import { withScriptjs, withGoogleMap } from 'react-google-maps';
import config from '../config';
import LocationContext from './LocationContext'
import InfoDisplay from './InfoDisplay'
import Map from './Map'
import './Home.scss';

const WrappedMap = withScriptjs(withGoogleMap(Map))

class Home extends Component {
    constructor(props) {
        super(props);
        this.myRef = React.createRef();
        this.state = {
            hasSelection: false
        }
    }

    static contextType = LocationContext

    setSelection = (boolean) => {
        this.state.hasSelection !== boolean && (
            this.setState({
                hasSelection: boolean
            })
        )   
    }
    
    render() {
        const contextValue = { 
            hasSelection: this.state.hasSelection,
            setSelection: this.setSelection,
        }

        // controls map size on selection (only on mobile)
        const isMobileDevice = window.innerWidth <= 1023
        const height = isMobileDevice && this.state.hasSelection ? "35vh" : "100vh"
        const width = !isMobileDevice && this.state.hasSelection ? "55vw" : "100vw"
        return (
            <LocationContext.Provider value={ contextValue }>
                <div className="map_container">
                    <div id="map" ref={this.myRef}>
                        <WrappedMap
                            googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${config.GOOGLE_MAPS_API_KEY}`}
                            loadingElement={<div className="loadingElement" style={{ height: height }}/>}
                            containerElement={<div className="containerElement" style={{ height: height }}/>}
                            mapElement={<div className="mapElement" style={{ height: height, width: width}}/>}
                        />
                    </div>
                </div>
            </LocationContext.Provider>
        );
    }
}

export default Home;