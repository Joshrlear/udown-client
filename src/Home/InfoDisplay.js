import React, { useContext, useEffect, useState } from 'react'
import config from '../config'
import io from 'socket.io-client'
import LocationContext from './LocationContext'
import udownContext from '../UdownContext'
import fetches from '../fetches'
import './InfoDisplay.scss'

let socket
socket = io(config.API_ENDPOINT)

export default function InfoDisplay(props) {
  const width = Math.ceil(window.innerWidth)
  const hieght = Math.ceil(window.innerHeight / 5)
  const location = useContext(LocationContext)
  const chatFuncs = useContext(udownContext)
  const [locationPhoto, setLocationPhoto] = useState(`https://via.placeholder.com/${width}x${hieght}`)

  function sendTxt() {
    const { chatOpened, startChat } = chatFuncs
    const { getOtherUsers } = fetches.profileFetches
    const { user_id } = localStorage

    // get user phone_number
    const phoneResult = Promise.resolve(getOtherUsers(user_id, 'phone_number'))
    phoneResult.then(value => {
      if (value) {
        value.map(user => console.log('getOtherUsers info',user))
        let otherUser
        const users = value.map(user => 
          otherUser = {
            'user_id': user.id,
            'phone': user.phone_number
          })
          console.log(users)
         const reqBody = { 
          'username': localStorage.username, 
          'location': location.name,
          'users': users
        }
        fetch(`${config.API_ENDPOINT}text`, {
              method: 'POST',
              body: JSON.stringify(reqBody),
              headers: {
                'Content-Type': 'application/json'
              }
            })
            .then(res => {
              if(!res.ok) {
                return res.json().then(err => {
                  throw err
                })
              }
              else {
                return res.json()
              }
            })
            .then(res => {
              const user = res.username
              const room = res.roomName
              const loc = {
                name: location.name,
                address: location.address
              }
              //socket.emit('join', { user, room, loc })
              !chatOpened && (startChat(room))
            })
            .catch(err => {
              console.log(err)
            })
      }
      else {console.log('no one has a number :(')}
    }) 
  }

  return (
      <div className={location.name === "Name" ? "info_display" : "info_display active"}>
        {/* <img src={location.photo || `https://via.placeholder.com/${width}x${hieght}`}/> */}
        <div className="info_container">
        <div className="button_container">
          <button className="playBtn" onClick={sendTxt}>PLAY</button>
        </div>
          <main className="info">
            <div className="header-button">
              <header>
                <h2>{location.name || "Location Name"}</h2>
                <p>Currently Available!</p>
              </header>
            </div>
            <section className="main_info">
                <p>{location.hours_of_operations}</p>
              <div className="address">
                <h3>Address:</h3>
                <p>{location.address || "Location Address"}</p>
              </div>
            </section>
          </main>
        </div>
      </div>
  )
}

