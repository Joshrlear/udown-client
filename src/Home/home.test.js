import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { withScriptjs, withGoogleMap } from 'react-google-maps';
import config from '../config';


import Home from './Home';
import InfoDisplay from './InfoDisplay';
import Map from './Map';

configure({ adapter: new Adapter() })

describe('Components', () => {
    const mappedComponents = [Home, InfoDisplay]
    mappedComponents.map(Component => {
        it(`renders without crashing`, () => {
            const div = document.createElement('div')
            ReactDOM.render(
            <BrowserRouter>
               <Component/>
            </BrowserRouter>, div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
    const WrappedMap = withScriptjs(withGoogleMap(Map))
    test("renders Map correctly", () => {
        const wrapper = shallow(<WrappedMap
            googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${config.GOOGLE_MAPS_API_KEY}`}
            loadingElement={<div className="loadingElement" style={{ height: '100%' }}/>}
            containerElement={<div className="containerElement" style={{ height: '100%' }}/>}
            mapElement={<div className="mapElement" style={{ height: '100%', width: '100%' }}/>}
        />);
      
        expect(wrapper).toMatchSnapshot();
    });
    const location = {
        name: 'park name',
        address: '123 main street',
        hours_of_operations: 'open'
    }

    it('renders Home correctly', () => {
        const wrapper = shallow(<Home/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders InfoDisplay with location properties', () => {
        const wrapper = shallow(<InfoDisplay location={ location }/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})