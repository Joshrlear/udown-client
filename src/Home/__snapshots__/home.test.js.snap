// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components renders Home correctly 1`] = `
<ContextProvider
  value={
    Object {
      "hasSelection": false,
      "setSelection": [Function],
    }
  }
>
  <div
    className="map_container"
  >
    <div
      id="map"
    >
      <withScriptjs(withGoogleMap(Map))
        containerElement={
          <div
            className="containerElement"
            style={
              Object {
                "height": "100vh",
              }
            }
          />
        }
        googleMapURL="https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=undefined"
        loadingElement={
          <div
            className="loadingElement"
            style={
              Object {
                "height": "100vh",
              }
            }
          />
        }
        mapElement={
          <div
            className="mapElement"
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          />
        }
      />
    </div>
  </div>
</ContextProvider>
`;

exports[`Components renders InfoDisplay with location properties 1`] = `
<div
  className="info_display active"
>
  <div
    className="info_container"
  >
    <div
      className="button_container"
    >
      <button
        className="playBtn"
        onClick={[Function]}
      >
        PLAY
      </button>
    </div>
    <main
      className="info"
    >
      <div
        className="header-button"
      >
        <header>
          <h2>
            Location Name
          </h2>
          <p>
            Currently Available!
          </p>
        </header>
      </div>
      <section
        className="main_info"
      >
        <p />
        <div
          className="address"
        >
          <h3>
            Address:
          </h3>
          <p>
            Location Address
          </p>
        </div>
      </section>
    </main>
  </div>
</div>
`;

exports[`Components renders Map correctly 1`] = `ShallowWrapper {}`;
